mesonbuild/backend/ninjabackend.py

```diff
@@ -1057,9 +1057,10 @@ def generate_target(self, target):
         self.add_build(elem)
         #In AIX, we archive shared libraries. If the instance is a shared library, we add a command to archive the shared library
         #object and create the build element.
-        if isinstance(target, build.SharedLibrary) and self.environment.machines[target.for_machine].is_aix():
-            elem = NinjaBuildElement(self.all_outputs, linker.get_archive_name(outname), 'AIX_LINKER', [outname])
-            self.add_build(elem)
+        if self.environment.machines[target.for_machine].is_aix() and isinstance(target, build.SharedLibrary):
```

Comment:
Random question: why is the order being reversed here? Was there something wrong with doing `isinstance(...) and self.environment.*` that this needed to be switched?

Reply:
> Random question: why is the order being reversed here? Was there something wrong with doing `isinstance(...) and self.environment.*` that this needed to be switched?

There was nothing wrong. I was just trying to see if that msys2 clang build failure had got to do something with it. So I was using the short circuit method of python to evaluate. So if the platform is not AIX, then it will not evaluate the second part of the condition and after that if that msys2 clang build failure passed (Though in reality I knew it does not matter) then I might had to think why that happened. 

So yeah, that is why I tried that.

Reply:
Then can you please restore it to its original state? :)

(And squash the Pull Request into one commit per mergeable, logical, change. In this case the entire PR should be one logical change -- changing the logic to use the .target_aix_so_archive property everywhere.)

Reply:
@eli-schwartz Here we go. Done :)

Thanks..